fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -1120,6 +1134,55 @@ protected RuntimeException mapAsyncToSyncException(@Nonnull Throwable ex) {
         return future.get();
     }
 
+    /**
+     * Log warning and close tracked contexts that have been open for too long.
+     * @param minAgeSeconds number of seconds above which to warn
+     * @return number of such contexts found
+     */
+    public int warnAndCloseOldTrackedOpenContexts(long minAgeSeconds) {
```

Comment:
Is this public so that a consumer might decide that they want to run this more often (or I suppose in their own tests)? It seems that in general, this should be package private, but I could also see the argument that either of those would be useful features.

Reply:
It needs to be public so that testing frameworks can call it when shutting down and find leaks, for instance. I added an annotation.

I added a KPI for when something is closed because it was open for too long.